code/modules/vehicles/pioneer_stone.dm

```diff
@@ -0,0 +1,63 @@
+/obj/vehicle/ridden/pioneer_stone
+	name = "pioneer stone"
+	desc = "Pioneer's used to ride these babies for miles"
+	icon_state = "pioneer_stone"
+	max_integrity = 500
+	armor = list("melee" = 75, "bullet" = 0, "laser" = 25, "energy" = 25, "bomb" = -75, "bio" = 100, "rad" = 100, "fire" = 100, "acid" = 100) //it is a fucking stone, what do you expect
+	are_legs_exposed = TRUE
+	var/max_damage_force = 25
+	
+	fall_off_if_missing_arms = TRUE
+	resistance_flags = FIRE_PROOF | UNACIDABLE | ACID_PROOF | FREEZE_PROOF
+
+/obj/vehicle/ridden/pioneer_stone/Initialize()
+	. = ..()
+	var/datum/component/riding/D = LoadComponent(/datum/component/riding)
+	D.set_riding_offsets(RIDING_OFFSET_ALL, list(TEXT_NORTH = list(0, 4), TEXT_SOUTH = list(0, 4), TEXT_EAST = list(0, 4), TEXT_WEST = list( 0, 4)))
+	D.vehicle_move_delay = 2
+
+/obj/vehicle/ridden/pioneer_stone/proc/Change_move_delay()
+	var/datum/component/riding/D = LoadComponent(/datum/component/riding)
+	var/turf/T = get_turf(src)
+	if(is_mining_level(T.z)) //pioneer stone works best on lavaland
+		D.vehicle_move_delay = 1
+	else
+		D.vehicle_move_delay = 2
+
+/obj/vehicle/ridden/pioneer_stone/Moved()
+	playsound(src, 'sound/effects/clang.ogg', 25, TRUE)
+	Change_move_delay()
+	. = ..()
+	
+
+
+/obj/vehicle/ridden/pioneer_stone/Bump(atom/movable/A)
+	. = ..()
+	
+	if(!(A.density && has_buckled_mobs())||(!ishuman(A)))
+		return FALSE
+	var/atom/throw_target = get_edge_target_turf(A, dir)
+	var/mob/living/carbon/human/H = A
+	var/damage_force = rand(10,max_damage_force)
+	H.apply_damage(damage_force, BRUTE)
+	var/turf/T = get_turf(src)
+	obj_integrity -= damage_force * is_mining_level(T.z)
```

Comment:
This means it won’t take damage off the mining z level
Is_mining_level will return 1 when true and 0 when false and anything * 0 is 0

Reply:
> This means it won’t take damage off the mining z level
> Is_mining_level will return 1 when true and 0 when false and anything * 0 is 0

intended

Reply:
so you can just run over everyone on station with no damage to the stone and thats intended?

Reply:
wait no , fuck there should be ! in there. my bad

Reply:
you arent supposed to take damage on lavaland
my bad my bad